skbio/metadata/tests/test_interval.py

```diff
@@ -446,6 +446,34 @@ def test_init_upper_bound_lt_lower_bound(self):
         with self.assertRaises(ValueError):
             IntervalMetadata(-1)
 
+    def test_upper_bound_is_none(self):
```

Comment:
Thank you for reworking the tests to use `__eq__`. By using this strategy, private attributes will only be used within `IntervalMetadata.__eq__` and not relied upon in the unit tests.